src/playlists.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+import SpotifyWebApi = require('spotify-web-api-node');
+
+export class PlaylistsProvider implements vscode.TreeDataProvider<PlaylistAndTracks> {
+
+  constructor(private spotify: SpotifyWebApi) {}
+
+  async getChildren(element?: PlaylistAndTracks): Promise<PlaylistAndTracks[] | undefined> {
+    if (element === undefined) {
+      return this.getPlaylistAndTracks(this.spotify);
+    }
+    return element.children;
+    
+  }
+  async getTreeItem(element: PlaylistAndTracks): Promise<PlaylistAndTracks> {
+    return element;
+  }
+
+  async getPlaylistAndTracks(spotify: SpotifyWebApi) {
+    try {
+      const user = await this.spotify.getMe();
+      const playlists = await this.spotify.getUserPlaylists(user.body.id);
+      return playlists.body.items.map((playlist: any) => new PlaylistAndTracks(playlist, this.spotify));
+      } catch (err: any) {
+      vscode.window.showErrorMessage(err);
+    }
+  }
+}
+
+
+export class PlaylistAndTracks extends vscode.TreeItem {
+  name: string;
+  artists: string;
+  uri: string;
+  children: Promise<PlaylistAndTracks[]> | undefined;
+  parent: PlaylistAndTracks | undefined;
+
+  constructor (public readonly data: any, spotify?: SpotifyWebApi, parent?: PlaylistAndTracks) {
+    super(data.name);
```

Comment:
Ah, now I know why I don't get any item labels for tracks. Try this, courtesy of the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator):

```suggestion
    super(data.track.name ?? data.name);
```

Reply:
Oooh this is gorgeous, didn't know about this!